lib/queuePopulator/KafkaLogConsumer/ListRecordStream.js

```diff
@@ -0,0 +1,82 @@
+const stream = require('stream');
+
+class ListRecordStream extends stream.Transform {
+    /**
+     * @constructor
+     * @param {Logger} logger logger
+     */
+    constructor(logger) {
+        super({ objectMode: true });
+        this._logger = logger;
+    }
+
+    /**
+     * Maps change stream operation type to
+     * backbeat supported types
+     * @param {string} operationType mongo opetation type
+     * @returns {string} supported operation type
+     */
+    _getType(operationType) {
+        switch (operationType) {
+            case 'insert':
+            case 'update':
+            case 'replace':
+                return 'put';
+            case 'delete':
+                return 'delete';
+            default:
+                this._logger.warn('Got unsupported operation', {
+                    method: 'ListRecordStream._getType',
+                    operationType,
+                });
+                return 'none';
+        }
+    }
+
+    /**
+     * parses json strings
+     * @param {string} blob sringified json
+     * @returns {Object} parsed json
+     */
+    _parseJson(blob) {
+        try {
+            let parsed = JSON.parse(blob);
+            if (typeof parsed === 'string') {
+                parsed = this._parseJson(parsed);
+            }
+            return parsed;
+        } catch (err) {
+            this._logger.warn('Got invalid kafka message value field format', {
+                method: 'ListRecordStream._parseJson',
+                value: blob,
+            });
+            return {};
+        }
+     }
+
+    /**
+     * Formats change stream entries
+     * @param {Object} data chunk of data
+     * @param {string} encoding enconding of data
+     * @param {Function} callback callback
+     * @returns {undefined}
+     */
+    _transform(data, encoding, callback) {
+        // data.value contains an over-stringified string which'll
+        // need to be parsed twice to get the object
+        const changeStreamDocument = this._parseJson(data.value);
+        const streamObject = {
+            timestamp: new Date(data.timestamp),
+            db: changeStreamDocument.ns && changeStreamDocument.ns.coll,
+            entries: [{
+                key: changeStreamDocument.documentKey && changeStreamDocument.documentKey._id,
+                type: this._getType(changeStreamDocument.operationType),
+                value: changeStreamDocument.fullDocument && changeStreamDocument.fullDocument.value,
+            }],
+        };
+        this.push(streamObject);
+        callback();
```

Comment:
I read that you can do this, ( I found it weird we were pushing in a transform function but that is what you need to do so I guess its right... )
```suggestion
        callback(null, streamObject);
```